lsp/src/Language/LSP/VFS.hs

```diff
@@ -92,6 +92,11 @@ data VirtualFile = VirtualFile
   -- remains in the map.
   , _file_text :: !Rope
   -- ^ The full contents of the document
+  , _language_id :: !(Maybe J.LanguageKind)
```

Comment:
You might also want an accessor function for this. It's a bit random but we have them for the text and version so let's do it for this too.

Reply:
sounds good!